### PR TITLE
fix(global setup): simplify ordering

### PR DESCRIPTION
**Pull request by Skn0tt**
As.discussed yesterday, this PR follows up on https://github.com/microsoft/playwright/pull/32955 with some ordering simplification. The new ordering is:

1. All `globalSetup` entries, in direct order
2. All callbacks returned from `globalSetup`, in reverse order
3. All `globalTeardown` entries, in direct order

Before we added support for multiple global setups, a failing callback meant that we wouldn't execute the teardown. This is broken by this change.

### CODE REVIEW

File: tests/playwright-test/global-setup.spec.ts [hunk 1]
```diff
@@ -410,19 +410,47 @@ test('globalSetup should support multiple', async ({ runInlineTest }) => {
...
+    'callback1',
+    'globalTeardown1',
     'globalTeardown2',
-    'globalSetup1Function',
-    // 'globalTeardown1' is missing, because globalSetup1Function errored out.
+  ]);
+  expect(result.output).toContain('Error: kaboom');
+});
+
+test('globalTeardown runs even if callback failed', async ({ runInlineTest }) => {
```
Comment (dgozman) [inline tests/playwright-test/global-setup.spec.ts:430]: I think this is already covered by `globalSetup1` from the previous test, isn't it?

**Review by dgozman — APPROVED** [review]

File: tests/playwright-test/global-setup.spec.ts (hunk 1 above)
Comment (Skn0tt) [reply tests/playwright-test/global-setup.spec.ts:430]: it is, yes. and it's also covered from the other test I had to touch. but this is the main breaking behaviour change, and I wanted to have an explicit test to cover this.

**Comment by Skn0tt** [comment]
@dgozman I had to change two more tests here in `runner.spec.ts` and `esm.spec.ts`. I'm a little concerned about the `runner.spec.ts` change: We'll start executing teardown even if setup failed. Let me know if you think this is fine.

**Comment by dgozman** [comment]
> @dgozman I had to change two more tests here in `runner.spec.ts` and `esm.spec.ts`. I'm a little concerned about the `runner.spec.ts` change: We'll start executing teardown even if setup failed. Let me know if you think this is fine.

Seems fine to me, we have explicitly agreed on this behavior.

**Pull request merged**